### DISCUSSION TITLE
Error message on dzsave of a larger high resolution (dpi 4800 svg file - could anybody please explain

### DISCUSSION THREAD

**Discussion by liveman1** [libvips/libvips#2295] [Q&A]
Hi,

I get following error message, when I try to create DZI from a svg file using
vips.exe dzsave sample\test.svg[dpi=4800] test.zip --layout dz --tile-size 506 --suffix .jpg[Q=70]

(vips.exe:40508): VIPS-WARNING **: 17:44:53.956: error in tile 0 x 0
(vips.exe:40508): VIPS-WARNING **: 17:44:54.254: error in tile 32767 x 0
(vips.exe:40508): VIPS-WARNING **: 17:44:54.390: error in tile 0 x 0
svgload: SVG rendering failed
svgload: SVG rendering failed

It does not happen, if I keep the DPI lower e.g. at 2400.

Could anybody please help me to understand this error message & what I could do to get a higher resolution DZI?

Thanks a lot!
Steffen

**Comment by jcupitt** [comment]
Hi @liveman1,

This looks like a regression in librsvg, it certainly used to work. 

I would try with an older libvips binary. I'll see if I can find a workaround.

**Comment by jcupitt** [comment]
It seems there's been a change in librsvg and you now need to call rsvg_handle_render_document() to render large SVGs. I've made an issue for the fix.

**Reply by liveman1** [reply]
Thanks a lot! You are awesome!
Do you have any hint which libvips binary might work?

**Reply by jcupitt** [reply]
Sorry, no idea, perhaps 8.9?

**Comment by jcupitt** [comment]
I tried a few versions of librsvg and it was working in 2.40.19. However there are quite a few rendering errors when the size goes over 32000 -- I suppose that's why they removed the feature.

Sorry, I don't think this is ever going to work.

**Reply by liveman1** [reply]
Ok, thanks. I have seen that this version of librsvg is used in libvips 8.9. I will give it another try.
Is there any other image format that works, when size goes over 32000 pixels?
Thanks

**Reply by jcupitt** [reply]
Other image format: it depends what your aim is. Could you explain what you are trying to achieve?

**Reply by liveman1** [reply]
I designed a huge svg with tons of details. I would like to visualize it on the web with all details. With the current dpi limitation I am not able to zoom in to visualize further details.
Which image format would you recommend?

**Reply by jcupitt** [reply]
Ah I see.

Perhaps you could crop the SVG into four in your editing package, render four 32757 x 32767 images, then join them and make the pyramid?

You probably need more than 64k x 64k. Maybe 4 x 4 tiles? Would 128k x 128k be enough?

**Reply by liveman1** [reply]
I would be very thankful if you could give me some guidance how to do this uing libvips.

**Reply by jcupitt** [reply]
Sorry, librsvg can't do this. You'll need to crop and save four (or 16 times) in your SVG editor. 

Are you using inkscape? There's probably a plugin for this (though I've not checked).

**Reply by jcupitt** [reply]
I bet this would work. Make a grid object, place it over your drawing, then use path divide.

https://inkscape.org/forums/questions/splitting-an-image-into-segments/

**Reply by liveman1** [reply]
It works perfectly with png files. So converting the pdf to a high-res png & then generating the dzi from there. I am quite happy now :)    Thanks a lot for your great help!
